docs/development-environment.md

```diff
@@ -1,298 +1,444 @@
-
-We provide a standard installation of WordPress for making Jetpack development easier.
-
-You can read the details in [docker/README.md](../docker/README.md)
-
-## Standard Development/debug Tools
-
-### WP_DEBUG
-
-You should do all Jetpack development with `define( 'WP_DEBUG', true );` in your `wp-config.php`, making sure that you’re not generating any Notices or other PHP issues in your error_log.
-
-### SCRIPT_DEBUG
-
-By default, WordPress loads minified versions of Jetpack's JS files. If you want to work with them, add `define( 'SCRIPT_DEBUG', true );` in your `wp-config.php`. This tells WordPress to load the non-minified JS version, allowing you to see your changes on page refresh. This applies to the JS files outside of `_inc/client/` and `extensions/`.
-
-### WP-CLI
+<a name="jetpack-linting-codestandards-code-linting"></a>
+## Linting
 
-Jetpack CLI is a command line interface for Jetpack, extending off of WP-CLI for WordPress. You can easily modify your installation of Jetpack with a just a few simple commands. All you need is SSH access and a basic understanding of command line tools.
+* ### Linting Jetpack's PHP code
 
-Usage:
-
-* `wp jetpack status [<full>]`
-* `wp jetpack module <list|activate|deactivate|toggle> [<module_name>]`
-* `wp jetpack options <list|get|delete|update> [<option_name>] [<option_value>]`
-* `wp jetpack protect <whitelist> [<ip|ip_low-ip_high|list|clear>]`
-* `wp jetpack reset <modules|options>`
-* `wp jetpack disconnect <blog|user> [<user_identifier>]`
-* `wp jetpack status`
-* `wp jetpack status [<full>]`
-
-More info can be found in [our support documentation](https://jetpack.com/support/jetpack-cli/).
-
-### JETPACK_DEV_DEBUG
-
-`JETPACK_DEV_DEBUG` constant can be used to enable development mode in Jetpack. Add `define( 'JETPACK_DEV_DEBUG', true );` in your `wp-config.php` to enable it. With Development Mode, features that do not require a connection to WordPress.com servers can be activated on a local WordPress installation for testing.
-
-Development mode automatically gets enabled if you don’t have a period in your site’s hostname, i.e. localhost. If you use a different URL, such as mycooltestsite.local, then you will need to define the `JETPACK_DEV_DEBUG` constant.
-
-You can also enable Jetpack’s development mode through a plugin, thanks to the jetpack_development_mode filter:
-
-`add_filter( 'jetpack_development_mode', '__return_true' );`
+	You can easily run these commands to set up all the rulesets and then lint Jetpack's PHP code. You need Composer to run this tool so check how to [install Composer](#installing-composer) if you don't have it yet.
+	
+	This will install all the CodeSniffer rulesets we need for linting Jetpack's PHP code. You may need to do this only once.
+	
+	```sh
+	$ composer install
+	```
+	
+	This runs the actual linting task.
+	
+	```sh
+	$ composer php:lint
+	```
+	
+	_There's also a handy `yarn php:lint` that will run `composer php:lint` if you prefer_.
```

Comment:
I wonder if we should keep only the Yarn version in here to keep things shorter?

Reply:
Makes sense, as it will reduce clutter. Updated.